feat(hujsonfmt): add basic hujsonfmt CLI command

jimeh opened the pull request:
Added as a separate Go module with its own go.mod file to avoid
polluting the root module's dependency tree.

The CLI interface and flags are very similar to gofmt:

    $ hujsonfmt -h
    usage: hujsonfmt [flags] [path ...]
      -d	display diffs instead of rewriting files
      -l	list files whose formatting differs from hujsonfmt's
      -m	minify results
      -s	standardize results to plain JSON
      -w	write result to (source) file instead of stdout

Given paths can either point directly to a file, or to a directory. In
the case of a directory, it will be recursively walked finding
all *.hujson files.

Input can also be provided via stdin if no paths are provided, or a
single path of "-" is provided:

    $ echo '{\n// hai\n"foo":"bar"}' | ./hujsonfmt
    {
    	// hai
    	"foo": "bar",
    }

knyar reviewed (changes requested):
Thank you for preparing this, and apologies that nobody got a chance to review this earlier.

jimeh commented:
@knyar Not a problem :)

Thanks for the feedback. I should have time to address in the next day or two.

creachadair commented on cmd/hujsonfmt/hujsonfmt.go:
```diff
@@ -0,0 +1,189 @@
+	case *list:
+		fmt.Println(filename)
+	case *write:
+		return os.WriteFile(filename, ouput, 0)
```
Perhaps you want https://pkg.go.dev/tailscale.com/atomicfile

jimeh replied:
I'm happy to use `atomicfile`, as long as the internal status of the package isn't an issue:

> This package should be considered internal; its API is not stable.

jimeh replied:
I decided to use a backup from rather than the `atomicfile` package, as I figured various tools/editors might not handle the original file being wholesale replaced with `os.Rename()`, rather than just updating the exiting file in place.

jimeh commented:
I believe I've addressed all the feedback. I also expanded input paths to support directories, in that it recursively walks a given directory finding all `*.hujson` files, just like `gofmt` walks given directories finding `*.go` files.

Personally, I'm somewhat unhappy about not having written any tests for this, but I believe it's probably simple enough to just about get away with it from an initial effort. That said, I'll happily provide some tests early after the holidays when my schedule is a bit more open :)

knyar commented:
Thank you!

dsnet commented:
This incurs a new dependency on an external module, which is highly unfortunate. It would be better if this could have been a different sub-module to keep the primary dependency tree lighter.

jimeh commented:
@dsnet apologies for the commit message style. I've been using conventional commits for so long now that it's practically a reflex.
